daemon/stop.go

```diff
@@ -70,7 +70,17 @@ func (daemon *Daemon) containerStop(container *containerpkg.Container, seconds i
 	}
 
 	// 2. Wait for the process to exit on its own
-	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(seconds)*time.Second)
+	var (
+		ctx    context.Context
+		cancel context.CancelFunc
+	)
+
+	if seconds >= 0 {
+		ctx, cancel = context.WithTimeout(context.Background(), time.Duration(seconds)*time.Second)
+	} else {
+		//if timeout is negative just wait until process exited by itself
+		ctx, cancel = context.WithCancel(context.Background())
```

Comment:
Do we need this cancel function?

I think we can simplify this a bit by just using `context.Background()` by default, and then setting the timeout ctx if seconds is >= 0. The `defer cancel()` can be in the timeout case, no need for the extra variable declaration.

Reply:
yeap! Maybe it's a better way.